lib/components/Atom/WheelPicker/base/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { HTMLAttributes, useEffect, useMemo, useRef } from "react";
+import clsx from "clsx";
+import { DropdownItem } from "@components/Atom/dropdown";
+import "./wheel-picker.scss";
+
+export interface WheelPickerProps extends HTMLAttributes<HTMLElement> {
+    data: { value: string | number }[];
+    selectedValue: string | number;
+    setSelectedValue: (value: string | number) => void;
+    handleKeyDown?: (e: React.KeyboardEvent) => void;
+    isFocused?: boolean;
+    dropDownItemClassName?: string;
+    containerClassName?: string;
+    listClassName?: string;
+    position?: "first" | "center" | "last" | undefined;
+}
+
+const WheelPicker = ({
+    data,
+    selectedValue,
+    setSelectedValue,
+    handleKeyDown,
+    isFocused,
+    dropDownItemClassName,
+    containerClassName,
+    listClassName,
+    position,
+    ...rest
+}: WheelPickerProps) => {
+    const itemsRefs = useRef<HTMLDivElement[]>([]);
+    const dataItemsMap = useMemo(
+        () =>
+            data.reduce(
+                (map, item, index) => map.set(item.value, index),
+                new Map(),
+            ),
+        [data],
+    );
+
+    const currentDataValue = useRef(dataItemsMap.get(selectedValue) ?? 0);
+    const itemsContainerRef = useRef<HTMLUListElement>(null);
+
+    const handleDataScroll = (event: Event) => {
+        if (!event.target) return;
+        const scrollTop = Math.max(
+            (event.target as HTMLUListElement).scrollTop,
+            0,
+        );
+        const selectedElement = Math.min(
+            Math.max(Math.floor(scrollTop / 48), 0),
+            data.length - 1,
+        );
+        (itemsRefs.current[selectedElement] as HTMLDivElement)?.focus({
+            preventScroll: true,
+        });
+        setSelectedValue(data[selectedElement].value);
+    };
+
+    const resizeObserver = new ResizeObserver(() => {
+        const index = Number(currentDataValue.current);
+        if (!itemsContainerRef?.current) return;
+        const divHeight = itemsContainerRef?.current.clientHeight;
+        itemsContainerRef.current.style.paddingTop = `${divHeight * 0.5}px`;
+        itemsContainerRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
```

Comment:
Can `0.5` be a variable multiplier so we can easily update later incase of UI change.